project_template/_base/project_name/apps/common/utils/redis.py

```diff
@@ -0,0 +1,35 @@
+import functools
+
+from django.conf import settings
+
+import redis
+
+
+@functools.lru_cache(maxsize=128)
```

Comment:
This code is not for caching Redis results - it's for caching instantiated Redis client itself. It will live in RAM.

This approach is used on Moneypark. @Quard please correct me if I'm wrong.

It could be useful if you use it a lot across the in module. For example:

```
from project.apps.common.utils.redis import redis_client


def some_logic_with_objects():
	...
	obj_one = redis_client().get('objects:one')
	...
	obj_two = redis_client().get('objects:two')


def some_logic_with_another_objects():
	...
	another_obj_one = redis_client().get('another:one')
	...
	another_obj_two = redis_client().get('another:two')


def do_something():
	res_a = some_logic_with_objects()
	res_b = some_logic_with_another_objects()

	redis_client().set('res:a', res_a)
	redis_client().set('res:b', res_b)
```

Yes, you right - it's very simple. But I guess that there are many people in company that haven't used Redis a lot (I mean directly - not as a Celery broker). And this small snippet will show them a good practice for naming keys in Redis using namespaces (instead of putting everything in root of DB by uuids and then receive a collision).

Reply:
Correct me if I wrong: We use `functools.lru_cache(maxsize=128)` to cache connected redis clients , and we cache up to 128 different `redis_connection_url` clients? And we'd like to do it because ... ?

Reply:
OK, my fault.
This is something like a connection pool, but Redis support this feature natively. 
https://github.com/andymccurdy/redis-py#connection-pools

In general, I see the very little value of having this and all related to Redis functionality in the skeleton. As some mystical snippets - yes.

Reply:
@SergeyKubrak 
no, we use `functools.lru_cache(maxsize=128)` to be able to have 128 maximum connected redis clients(connected instances) based on function params without necessary to create new each time when we call this function

@romanosipenko 
yes, redis-py support pooling but we still have to have client instance to use it instead of creating a new instance each time

To use pool approach I prefer to have a singleton context manager what will have clearer API then the function that returns pool

```
with MyRedis().acuire() as redis:
    ...
```